catchup/service.go

```diff
@@ -36,21 +37,25 @@ import (
 )
 
 const catchupPeersForSync = 10
+const blockQueryPeerLimit = 10
 
 // this should be at least the number of relays
 const catchupRetryLimit = 500
 
+// PendingUnmatchedCertificate is a single certificate that is being waited upon to have it's corresponding block fetched.
+type PendingUnmatchedCertificate struct {
+	Cert         agreement.Certificate
+	VoteVerifier *agreement.AsyncVoteVerifier
+}
+
 // Ledger represents the interface of a block database which the
 // catchup server should interact with.
 type Ledger interface {
-	NextRound() basics.Round
-	LastRound() basics.Round
-	Wait(basics.Round) chan struct{}
+	agreement.LedgerReader
```

Comment:
Do you need the entire `LedgerReader` here as a dependency? (Or just `Wait`?)

Reply:
Needed for the usage of Certificate.Authenticate() call; I'm not a big fan of having yet-another Ledger interface.
I'd rather to have an instance of the data/Ledger directly, but I don't want to make this into a bigger change in the context of this change. ( I guess it used to be important for preventing circular referencing ? )

Reply:
I guess the point of this interface was to decouple the `Ledger`-related requirements of the `catchup` package from the rest of the codebase, and importing `agreement.LedgerReader` seems to work against that objective. I don't have strong opinions here, though.